html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/OAuth/OpenID.pm

```diff
@@ -17,6 +17,29 @@ has '+source' => (isa => 'pf::Authentication::Source::OpenIDSource');
 
 has 'token_scheme' => (is => 'rw', default => sub {"auth-header:Bearer"});
 
+=head2 _extract_username_from_response
+
+Extract the username from the response of the provider
+
+=cut
+
+sub _extract_username_from_response {
+    my ($self, $info) = @_;
+    my $source = $self->source;
+    my $username = $source->username_attribute;
+    if (!exists $info->{$username}) {
+        return $self->SUPER::_extract_username_from_response($info);
+    }
+
+    return $info->{$source->username_attribute} // $self->SUPER::_extract_username_from_response($info);
+}
+
+sub auth_source_params_child {
+    my ($self) = @_;
+    my $info = person_view($self->username());
```

Comment:
I got following error message when testing basic OpenID workflow:
```
Caught exception in captiveportal::Controller::Root->dynamic_application
"Undefined subroutine
&captiveportal::PacketFence::DynamicRouting::Module::Authentication::OAuth::OpenID::person_view
called at
/usr/local/pf/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/OAuth/OpenID.pm
line 39."
```
but `pf::person` is used here: https://github.com/inverse-inc/packetfence/blob/a60ff1f00e2a929377ffaeed39f27910909aa87d/html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication.pm#L20

Reply:
if I use `use pf::person` directly in `html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/OAuth/OpenID.pm`, it works.

Reply:
I push a fix for that error.